server/buildbuddy_server/buildbuddy_server.go

```diff
@@ -886,6 +888,48 @@ func (s *BuildBuddyServer) GetUsage(ctx context.Context, req *usagepb.GetUsageRe
 	return nil, status.UnimplementedError("Not implemented")
 }
 
+func (s *BuildBuddyServer) GetCacheMetadata(ctx context.Context, req *capb.GetCacheMetadataRequest) (*capb.GetCacheMetadataResponse, error) {
+	ctx, err := prefix.AttachUserPrefixToContext(ctx, s.env)
+	if err != nil {
+		return nil, err
+	}
+
+	resourceName := req.GetResourceName()
+	cacheType, err := ProtoCacheTypeToCacheType(resourceName.GetCacheType())
+	if err != nil {
+		return nil, err
+	}
+	cache, err := s.env.GetCache().WithIsolation(ctx, cacheType, resourceName.GetInstanceName())
+	if err != nil {
+		return nil, err
+	}
+
+	metadata, err := cache.Metadata(ctx, resourceName.GetDigest())
+	if err != nil {
+		if status.IsNotFoundError(err) {
+			return nil, nil
+		}
+		return nil, err
+	}
+
+	return &capb.GetCacheMetadataResponse{
+		SizeBytes:      metadata.SizeBytes,
+		LastAccessUsec: metadata.LastAccessTimeUsec,
+		LastModifyUsec: metadata.LastModifyTimeUsec,
+	}, nil
+}
+
+func ProtoCacheTypeToCacheType(cacheType resource.CacheType) (interfaces.CacheType, error) {
+	switch cacheType {
+	case resource.CacheType_AC:
+		return interfaces.ActionCacheType, nil
+	case resource.CacheType_CAS:
+		return interfaces.CASCacheType, nil
+	default:
+		return interfaces.UnknownCacheType, status.InvalidArgumentErrorf("unknown cache type %v", cacheType)
+	}
+}
```

Comment:
can be a separate PR, but it would be cleaner if we removed the cache type enums from interfaces.go and used the protos everywhere

Reply:
Yeah when I refactor things to use the ResourceName struct I'll clean that up too. We also have a third file that has a cache type enum but the numbers are switched... Very confusing...